airflow/api_fastapi/execution_api/routes/xcoms.py

```diff
@@ -215,7 +216,8 @@ def set_xcom(
 
     # TODO: This is in-efficient. We json.loads it here for BaseXCom.set to then json.dump it!
     try:
-        json.loads(value)
+        # Keep this as long as we do BaseXCom.set(), it serialises again
```

Comment:
Does BaseXCom.set do the serialization itself, or is that handled internally by SQLA because we are putting it in the an JSON attr? If so then this comment should be tweaked, as it's not BaseXCom.set that is serializing it again.